getMultiStarts.m

```diff
@@ -222,40 +234,51 @@
         error_count = 0;
         
         % Test evaluation of objective function at starting point
+        % Only for multi-start local, since other optimizers use a 
+        % different initialization
         if (strcmp(options.localOptimizer, 'fmincon'))
             if (strcmp(options.localOptimizerOptions.Hessian, 'on'))
-                [J_0,~,~] = negLogPostWErrorCount(parameters.MS.par0(:,i)); % objectiveWrapWErrorCount(parameters.MS.par0(:,i),objective_function,options.obj_type,options.objOutNumber);
+                % We have to check Hessian AND gradient computation, since
+                % simulation can give different results if we're close to
+                % numerical failure
+                [J_0,~,~] = negLogPostWErrorCount(parameters.MS.par0(:,iMS));
```

Comment:
Hum...
I see your point.

fmincon calls the objective and the hessian function with the same parameter vector. So the initial call must work for both. Taking the max should only ensure that both of them work.

The problem is currently the following: For me, the fix made PESTO work when doing an optimization with 200 multistarts, so it was quite helpful in my case. And the objective function is not necessarily unstable, it may just be that one initial point out of some hundreds was chosen unluckily, at the very border of what is still integrable. It is frustrating to see that you computation failed at point 4 or 5 out of some hundreds, when looking at it after one or two days, if this can be cured this easily.

Anyway, this double call is done only if you use a user provided Hessian matrix. One additional gradient call should hopefully not kill you, when you call the Hessian many times...